tf_agents/environments/wrappers.py

```diff
@@ -719,3 +719,45 @@ def _step(self, action):
 
     time_step = self._env.step(action)
     return self._add_history(time_step, action)
+
+
+@gin.configurable
+class MultiDiscreteToDiscreteWrapper(PyEnvironmentBaseWrapper):
+  """Converts an environment with Multidiscrete action space into a Discrete one."""
+
+  def __init__(self, env):
+    """Creates an environment wrapper that converts the action space.
+
+    Args:
+      env: Environment to wrap.
+    """
+    super(MultiDiscreteToDiscreteWrapper, self).__init__(env)
+
+    self._multidiscrete_spec = env.action_spec()
+    self._discrete_spec = array_spec.BoundedArraySpec(
+      shape=(np.prod(env.action_spec().shape),),
+      dtype=env.action_spec().dtype,
+      minimum=0,
+      maximum=np.prod(env.action_spec().shape),
+    )
+
+  def _discrete_to_multidiscrete(self, discrete_action):
+    """Convert discrete action to multidiscrete action from original env.
+
+    The conversion follows lexicographical order.
```

Comment:
Why lexicographical order? I think the mapping may be clearer if it simply iterated over the multiple discrete spaces.